central/postgres/pruning.go

```diff
@@ -47,6 +48,42 @@ const (
 		(SELECT 1 FROM pods WHERE pi.poduid = pods.Id)) 
 		delete FROM process_indicators pi USING orphan_proc op WHERE pi.id = op.id AND 	
 		(signal_time < now() AT time zone 'utc' - INTERVAL '%d MINUTES' OR signal_time IS NULL)`
+
+	// (snapshots.reportstatus_runstate = 3 OR snapshots.reportstatus_runstate = 5)
+	// ...gives us the report jobs that are in final state.
+	//
+	// (SELECT MAX(latest.reportstatus_completedat) FROM ` + schema.ReportSnapshotsTableName + ` latest
+	// WHERE latest.reportstatus_completedat IS NOT NULL
+	// AND snapshots.reportconfigurationid = latest.reportconfigurationid
+	// AND latest.reportstatus_runstate = 5
+	// GROUP BY latest.reportstatus_reportnotificationmethod, latest.reportstatus_reportrequesttype)
+	//	...gives us the last successful report job for each config, for each notificated method, and each request type.
+	//
+	// (SELECT 1 FROM ` + schema.BlobsTableName + ` blobs
+	//	WHERE blobs.name not ilike '%/snapshots.reportid')
+	// ...tells us if the report still exists in the blob store.
+	//
+	// (reportstatus_completedat < now() AT time zone 'utc' - INTERVAL '%d MINUTES')
+	// ...gives us the reports that are outside the retention window.
+	pruneOldReportHistory = `DELETE FROM ` + schema.ReportSnapshotsTableName + ` WHERE reportid IN
+		(
+			SELECT snapshots.reportid FROM ` + schema.ReportSnapshotsTableName + ` snapshots
+			WHERE (snapshots.reportstatus_runstate = 3 OR snapshots.reportstatus_runstate = 5) 
+			AND snapshots.reportstatus_completedat NOT IN
+			(
+				SELECT MAX(latest.reportstatus_completedat) FROM ` + schema.ReportSnapshotsTableName + ` latest 
+				WHERE latest.reportstatus_completedat IS NOT NULL
+				AND snapshots.reportconfigurationid = latest.reportconfigurationid
+				AND latest.reportstatus_runstate = 5
+				GROUP BY latest.reportstatus_reportnotificationmethod, latest.reportstatus_reportrequesttype
```

Comment:
We just need to preserve the last successful scheduled job for each config because the CVE filter option is `sinceLastSentScheduledReport` . So this can be simplified.

Reply:
I have revised my initial design with the objective of ensuring that customers can easily comprehend the mechanics involved. The earlier design contained intricate conditions, which could potentially lead to complexities. By retaining the concept of preserving the last job for each type, customers can now obtain the last job status outcome across various categories they opt for.

Reply:
That makes sense.